ci/build_cpp.sh

```diff
@@ -13,11 +13,11 @@ export CMAKE_GENERATOR=Ninja
 
 rapids-print-env
 
-version=$(rapids-generate-version)
+rapids-generate-version > ./VERSION
```

Comment:
In the conda case where we need the value again to set the env var for the conda build I don't think piping then reading back from file is an improvement over the original approach. I won't block merging over that though, up to you if you like this better or if you don't but don't think it's worth another iteration.

Reply:
Ha I was anticipating a review comment of the form "what about `build_cpp.sh`, that's now inconsistent with the others" if it wasn't changed this way.

I'm gonna push another commit anyway in response to https://github.com/rapidsai/rmm/pull/1502#discussion_r1612462557, so I'll revert this.

Reply:
reverted in https://github.com/rapidsai/rmm/pull/1502/commits/7aeba946df9a5c50f513e9516afb33c27a5c0435.